frontend/src/components/common/Typography.tsx

```diff
@@ -0,0 +1,49 @@
+import { PropsWithChildren, useState } from 'react'
+
+import { Typography as MaterialTypography, Theme, TypographyProps } from '@material-ui/core'
+import { createStyles, makeStyles } from '@material-ui/core/styles'
+import LinkIcon from '@material-ui/icons/Link'
+
+import LinkIconButton from 'components/common/LinkIconButton'
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    linkIcon: {
+      transform: 'rotate(-45deg)',
+      color: theme.palette.primary.main,
+      '&:hover': {
+        color: theme.palette.primary.dark,
+      },
+    },
+    hideLinkIcon: {
+      visibility: 'hidden',
+    },
+  }),
+)
+
+type TypographyParams = PropsWithChildren<TypographyProps> & {
+  id?: string
+  linkable?: boolean
+  component?: string
```

Comment:
This looks like a leftover that we don't use

Reply:
Hello, @kachar , that it is interesting one, it look like `component` is not part of `TypographyProps`, it does not work without adding manually `component` prop.  You can check how I implemented it [here](https://github.com/podkrepi-bg/frontend/pull/278/commits/2acb40725303920a7176445428506f95466b2249)

Reply:
I was observing the `component` prop at their api docs https://material-ui.com/api/typography/

It's defined in a parent component of `Typography` - `OverridableComponent`

You can see the definition at:

https://github.com/mui-org/material-ui/blob/14d95615dd0b31c616abb760a29ec30a8fa9005f/packages/material-ui/src/OverridableComponent.d.ts#L12-L18

Maybe we have a way to inherit it

Reply:
Found that other people are experiencing this issue too - https://github.com/mui-org/material-ui/issues/19512

TLDR;
`component` is not defined in `TypographyProps` issue is closed - waiting for a new report on v5 of Material-UI.

What do you think about using `component: React.ElementType` and leaving it for now?

Reply:
Let's leave it for now `component: React.ElementType`, it's a good solution without making it a strict `string` type

Thanks for the research :man_scientist: :microscope: